client/state/jetpack-remote-install/test/reducer.js

```diff
@@ -0,0 +1,89 @@
+/** @format */
+/**
+ * Internal dependencies
+ */
+import reducer, { error, inProgress, isComplete } from '../reducer';
+import {
+	jetpackRemoteInstall,
+	jetpackRemoteInstallComplete,
+	jetpackRemoteInstallUpdateError,
+} from '../actions';
+
+const url = 'https://yourgroovydomain.com';
+const errorCode = 'INVALID_CREDENTIALS';
+
+describe( 'reducer', () => {
+	test( 'should export expected reducer keys', () => {
+		const state = reducer( undefined, {} );
+
+		expect( state ).toHaveProperty( 'isComplete' );
+		expect( state ).toHaveProperty( 'inProgress' );
+		expect( state ).toHaveProperty( 'error' );
+	} );
+
+	describe( 'isComplete', () => {
+		test( 'should store install complete by url', () => {
+			const state = isComplete( undefined, jetpackRemoteInstallComplete( url ) );
+			expect( state[ url ] ).toBe( true );
+		} );
+
+		test( 'should be cleared on install request', () => {
+			const initialState = {
+				[ url ]: {
+					isComplete: true,
+				},
+			};
+			const state = isComplete( initialState, jetpackRemoteInstall( url ) );
+			expect( state[ url ] ).toBeFalsy();
+		} );
+	} );
+
+	describe( 'inProgress', () => {
+		const inProgressState = {
+			[ url ]: {
+				inProgress: true,
+			},
+		};
+
+		test( 'should be set on install request', () => {
+			const state = inProgress( undefined, jetpackRemoteInstall( url ) );
+			expect( state[ url ] ).toBeTruthy();
```

Comment:
What's the motivation for truthy/falsy tests?

Unless there's an argument against, I'd prefer more precision:

```js
expect().not.toBeDefined()
expect().toBe( true )
```

Reply:
I don't feel strongly about this, but my thinking was that consumers don't care at all about the underlying mechanism, for example whether or not the indexed reducer deletes the whole index. The tests weren't even consistent in this, though, so changing anyway.